src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3682,8 +3682,8 @@ private void EmitConditionalOperator(BoundConditionalOperator expr, bool used)
             // Generate branchless IL for (b ? 1 : 0).
             if (used && _ilEmitStyle != ILEmitStyle.Debug &&
                 (IsNumeric(expr.Type) || expr.Type.PrimitiveTypeCode == Cci.PrimitiveTypeCode.Boolean) &&
-                hasIntegralValueZeroOrOne(expr.Consequence, out var isConsequenceOne) &&
-                hasIntegralValueZeroOrOne(expr.Alternative, out var isAlternativeOne) &&
+                isIntegralValueZeroOrOne(expr.Consequence) is bool isConsequenceOne &&
```

Comment:
>isIntegralValueZeroOrOne(expr.Consequence) is bool isConsequenceOne && [](http://example.com/codeflow?start=16&length=70)

Involvement of nullable values and patterns makes the code significantly harder to follow, in my opinion. We always know if an expression is a constant zero or a constant one, there is no "unknown" state in this sense. I suggest reverting changes in this `if` statement and to the signature of the helper.